java/code/src/com/suse/manager/webui/services/impl/test/SaltServiceTest.java

```diff
@@ -139,6 +142,48 @@ public void testGenerateSSHKeyExists() throws IOException {
         systemQuery.close();
     }
 
+    @Test
+    public void testLogEntryWithSensitiveData() {
+
+        var sensitiveDataRegex =
+                "-----BEGIN CERTIFICATE-----" +
+                        "|" +
+                        "-----BEGIN RSA PRIVATE KEY-----";
+
+        var saltClient = mock(SaltClient.class);
+        var saltService = new SaltService(saltClient);
+        var listAppender = SaltTestUtils.enableTestLogging(SaltService.class);
+
+        context().checking(new Expectations() {{
+            oneOf(saltClient).call(
+                    with(SaltTestUtils.functionEquals("mgrutil", "check_ssl_cert")),
+                    with(any(Client.class)),
+                    with(any(Optional.class)),
+                    with(any(Map.class)),
+                    with(any(TypeToken.class)),
+                    with(any(AuthMethod.class))
+            );
+            will(returnValue(SaltTestUtils.getCompletionStage(
+                    "/com/suse/manager/webui/services/impl/test/service/ssh_cert_check.json",
+                    new TypeToken<JsonParsingError>() {
+                    }.getType())));
+        }});
+
+        var call = MgrUtilRunner.checkSSLCert(
+                "Certificate:\ncert\n-----BEGIN CERTIFICATE-----\ncert\n-----END CERTIFICATE-----",
+                new SSLCertPair(
+                        "-----BEGIN CERTIFICATE-----\ncert\n-----END CERTIFICATE-----",
+                        "-----BEGIN RSA PRIVATE KEY-----\nkey\n-----END RSA PRIVATE KEY-----"
+                ),
+                null
+        );
+        var result = saltService.callSync(call);
+
+        assertEquals(result, Optional.empty());
+        assertFalse(listAppender.getLog().isEmpty());
+        assertFalse(listAppender.matchInLogs(sensitiveDataRegex));
```

Comment:
Rather than matching the same regex than the code I would try to look to the content of the certificates here to ensure they are replaced by `HIDDEN`

Reply:
Oh that makes total sense